### PR TITLE
Build: Enforce spotless & spotlessApply

### PR DESCRIPTION
**Pull request by nastra**
Docs are going to be updated as part of https://github.com/apache/iceberg-docs/pull/125

### CODE REVIEW

File: .baseline/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -495,7 +437,6 @@
             <property name="allowOneCharVarInForLoop" value="true"/>
             <message key="name.invalidPattern" value="Local variable name ''{0}'' must match pattern ''{1}''."/>
         </module>
-        <module name="MethodLength"/> <!-- Java Coding Guide: Methods and functions: focused, crisp, concise -->
```
[the remark this answers is not in the document]
Comment (nastra) [reply .baseline/checkstyle/checkstyle.xml]: reverted this change and added suppressions to the few places that had sligthly longer methods

**Review by dimas-b — APPROVED** [review]

**Comment by nastra** [comment]
@rdblue I think it would be great if those commits don't get squashed as it makes sense to have them separate

**Comment by rdblue** [comment]
Merged! I did the rebase and merge since the `spotlessApply` commit is so huge.

**Comment by zinking** [comment]
@nastra  why the default width is required to be changed?  you are causing so much trouble for downstream to merge these conflicts.

**Comment by nastra** [comment]
@zinking we had to pick one format that produce the same format and can be enforced across cmd line / Eclipse / Intellij and we went with the *google-java-format*, which has a line length of 100 (and is not customizable unfortunately). 
We had to do this short-term disruption for the sake of long-term consistency and to overall improve the project. For some context about why Spotless + *google-java-format* was chosen, please refer to the [ML thread](https://lists.apache.org/thread/rkv5omc5hfpyx9w87frhdyggbgvpv6sw).

In order to reduce the amount of conflicts, please follow the below steps:

1. Rebase your PR branch against the commit before the „Big Bang“
2. Run `./gradlew spotlessApply`
3. Squash all commits
4. Rebase against the latest HEAD of the master branch
5. Force-push your branch

Additionally, make sure to install the *google-java-format* plugin for your IDE as outlined in https://github.com/apache/iceberg-docs/pull/125 so that you can format code inside the IDE as well.

**Comment by wypoon** [comment]
I suggest the following order:
1. Squash all commits (if you're going to do this, might as well do it first).
2. Rebase against master at commit before [99b41eb](https://github.com/apache/iceberg/commit/99b41eb9fbae1eb6c0501ce1fd1c9008893583b4) (otherwise build will fail due to spotless being enforced) -- I normally build after the rebase to make sure the rebase is correct.
3. Rebase against HEAD of master.
4. Run `./gradlew spotlessApply`. I would build after that.

If we are going to rebase against master twice, before the spotlessApply ("Big Bang") and then to HEAD, in the first rebase, the code base has not yet been updated to the new style, so running `./gradlew spotlessApply` after this rebase means you're going to change a ton of files (correct me if I'm wrong, but `./gradlew spotlessApply` would apply to everything, right?). Why do this when you're going to rebase again on HEAD where the "Big Bang" has happened? I'd think you want to run `./gradlew spotlessApply` at a point where it'd at most modify the files that are added/modified in your change.

**Comment by nastra** [comment]
> I suggest the following order:
> 
> 1. Squash all commits (if you're going to do this, might as well do it first).
> 2. Rebase against master at commit before [99b41eb](https://github.com/apache/iceberg/commit/99b41eb9fbae1eb6c0501ce1fd1c9008893583b4) (otherwise build will fail due to spotless being enforced) -- I normally build after the rebase to make sure the rebase is correct.
> 3. Rebase against HEAD of master.
> 4. Run `./gradlew spotlessApply`. I would build after that.
> 
> If we are going to rebase against master twice, before the spotlessApply ("Big Bang") and then to HEAD, in the first rebase, the code base has not yet been updated to the new style, so running `./gradlew spotlessApply` after this rebase means you're going to change a ton of files (correct me if I'm wrong, but `./gradlew spotlessApply` would apply to everything, right?). Why do this when you're going to rebase again on HEAD where the "Big Bang" has happened? I'd think you want to run `./gradlew spotlessApply` at a point where it'd at most modify the files that are added/modified in your change.

The idea with squashing is to include all the changes from running `./gradlew spotlessApply` locally, so yes, this will change a ton of unrelated files, including the ones that are being modified for a PR. If you add those formatting changes into your commit and then do a rebase against latest `master`, then the only diff that should be left is the actual files that have been modified for that particular PR. 
This should minimize the amount of merge conflicts to an absolute minimum.

On the other hand, if you rebase changes (that are formatted with the old style) against latest `master` before locally applying `./gradlew spotlessApply`, then you'll certainly will have more conflicts (consisting of style + code changes).

**Pull request merged**